### ISSUE TITLE
Think `no-new-require` should be a recommendation

### ISSUE THREAD

**Issue by jimmywarting** [eslint-community/eslint-plugin-n#104]
...why? refactoring cjs to esm gets harder. 
it's a footgun for ppl writing this kind of syntax where they do `new require('app-header')` or stuff like 
`app = require('express')()`

chaining stuff on the `require()` is bad. 
it's better / easier if it looks more like ESM syntax and follows this format `const { xyz } = require(x)`
it's not recommended that you do things such as `require(x).foo().bar`

ppl complain that refactoring a project to esm is hard and takes long time. but having `no-new-require` as a recommendation would make it easier for ppl by spending less time and also making PR/reviews easier.

**Comment by voxpelli** [comment]
`const { xyz } = require(x)` also maps badly to ESM as it would need to be rewritten to:

```js
import x from 'x'
const { xyz } = x
```

An `import { xyz } from 'x'` would require `xyz` to be exported individually in an ESM file, a CJS file behaves as having `module.exports` exported as the `default` and ESM imports don't destructure objects, they pick individual exports.

That said, I'm neither pro or con on this.

**Comment by jimmywarting** [comment]
The statement `const { xyz } = require(x)` is somewhat similar to `import { xyz } from 'x'`, which could make it worthwhile. However, the ease of transitioning depends on how the module importing are exporting them ofc.

When the module is exported in a way similar to combining the [grouping](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/group-exports.md) and [export last](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/exports-last.md) rules, a code refactor becomes as simple as a one-line diff:

```diff
- module.exports = {
+ export = {
    x, 
    y,
    z
  }

With this export approach, refactoring becomes straightforward, and `import { xyz } from 'x'` comes closest to using ESM syntax without any breaking changes where the transition from cjs to esm are very high. Thus, combining these rules makes transitioning from CJS to ESM much easier to refactor at a later stage. 

However, certain cases, like `app = require('express')()`, `debug = require('debug')('webtorrent')`, and `stat = require('node:fs').stat,` are more troublesome to refactor and time-consuming. Therefore, the recommendation is to stick to only destructuring require statements, as it is the most sensible approach for handling them by treating them more or less similar to how ESM syntax closely resembles.

**Comment by scagood** [comment]
First, I think I agree that `no-new-require` should be recommended as I don't think it should be used in node. It ultimatly leads to misunderstandings in the code.

---

Secondly, I think you're talking about a possible new rule, something like `no-nested-require` (I am terrible at names, and open to suggestions 😂)

Here is a quick astexplorer example:
https://astexplorer.net/#/gist/1e0ae8545cf92ad0b8e6f45822bed6ae/ffbfdafbfd4f6a33b7c664e9e996ffaaeaa76124

<details>
<summary>👨‍💻 Here is the long texty example:</summary>

The rule could be something like this:
```js
export const meta = {
  type: "problem",
  hasSuggestions: true,
  fixable: false
};

export function create(context) {
  console.info(context);
  return {
    ':not(VariableDeclarator, ExpressionStatement) > CallExpression[callee.name="require"]'(node) {
      context.report({
        node: node,
        message: `Unexpected nested require`
      });
    }
  };
}
```

Given:
```js
require('dotenv');

const app1 = require('express')();

const express = require('express');
const app2 = express();

const stat = require('node:fs').stat;
```

You get:
```js
// Unexpected nested require (at 3:14)
   const app1 = require('express')();
// -------------^

// Unexpected nested require (at 8:14)
   const stat = require('node:fs').stat;
// -------------^
```

</details>
